app/models/pet_application.rb

```diff
@@ -1,4 +1,8 @@
 class PetApplication < ApplicationRecord
   belongs_to :application
   belongs_to :pet
+
+  def self.by_application_id(application_id)
```

Comment:
This is an interesting method. I think this is an efficient way to be able to query whether or not the id matches the application id we are looking for